### PR TITLE
[SLP]Add better minbitwidth analysis for udiv/urem instructions.

### PR DESCRIPTION
**Pull request by alexey-bataev**
Adds improved bitwidth analysis for udiv/urem instructions. The
analysis is based on similar version in InstCombiner.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -14165,6 +14165,28 @@ bool BoUpSLP::collectValuesToDemote(
...
+    if (!AttemptCheckBitwidth(
+            [&](unsigned BitWidth, unsigned OrigBitWidth) {
+              assert(BitWidth <= OrigBitWidth && "Unexpected bitwidths!");
+              APInt Mask = APInt::getBitsSetFrom(OrigBitWidth, BitWidth);
+              return MaskedValueIsZero(I->getOperand(0), Mask,
+                                       SimplifyQuery(*DL)) &&
+                     MaskedValueIsZero(I->getOperand(1), Mask,
+                                       SimplifyQuery(*DL));
+            },
+            NeedToExit))
```
Comment (RKSimon) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:14182]: This is starting to have a lot of code similar to canEvaluateTruncated in InstCombineCasts.cpp - can we reuse that code somehow - move it to ValueTracking.h or something?

Comment (alexey-bataev) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:14182]: Yeah, I thought about this. Actually, only the code for shifts/div/rem is duplicated, all other code is unique and/or requires some extra processing, which canEvaluateTruncated does not have. We can think about it later.

**Comment by alexey-bataev** [comment]
Ping!

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**